### ISSUE TITLE
getSignedUrl() - Custom domain name?

### ISSUE THREAD

**Issue by aichholzer** [aws/aws-sdk-js#891]
`Question`

Consider this request, for example:

```
params = {
  Bucket: 'myBucket,
  Key: 'image.jpg',
  ContentType: 'image/jpeg',
  Expires: 30
};

S3.getSignedUrl('putObject', params, function (error, url) {
  console.log(url);
});
```

which outputs: `https://myBucket.s3.eu-central-1.amazonaws.com/image.jpg?...`

I have created a `CNAME`; `media.domain.com` which points to `myBucket.s3.eu-central-1.amazonaws.com`

The idea here is to be able to provide a `customDomain` parameter where `media.domain.com` would be its value. This in order to have `getSignedUrl()` output `https://media.domain.com/image.jpg?...`

Just replacing the domain after `getSignedUrl()` does not work since the actual AWS S3 domain is also used for the signature.

In short; I would like to be able to specify a custom domain for putting objects to my S3 bucket.

Any ideas?

**Comment by LiuJoyceC** [comment]
Hi @aichholzer 
Thanks for your question.
You can set your bucket endpoint when you create your instance of S3: `var S3 = new AWS.S3({endpoint: 'media.domain.com', s3BucketEndpoint: true});` Then the url returned from `getSignedUrl` will be set to the specified endpoint. For a more detailed description of these params, please see the Constructor Details section of the API Docs: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html

Let me know if you have any further questions.

**Comment by aichholzer** [comment]
Hey @LiuJoyceC 
Thank you for your reply.
It worked, however, I had to do a little more research since the region my bucket is in only supports `v4` signatures and `signatureVersion` must be properly set in the configuration.

http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingAWSSDK.html#specify-signature-version

**Comment by nalbion** [comment]
@LiuJoyceC #1753 is related, but has extra complexity introduced by a CloudFront path pattern
